### ISSUE TITLE
Should not send content-type=application/xml when content is empty

### ISSUE THREAD

**Issue by ayeltsov** [Azure/Azurite#390]
### Which service(blob, file, queue, table) does this issue concern?
Blob
### Which version of the Azurite was used?
3.4.0
### Where do you get Azurite? (npm, DockerHub, NuGet, Visual Studio Code Extension)
npm
### What's the Node.js version?
v12.14.1
### What problem was encountered?
When blob is not found, Azurite emulator returns no response, yet it sends content-type application/xml. 
This breaks native code library that SQL Server uses.
There are a few discussions on the topic and general consensus is "do not declare content type as xml if it's not", for example [What should the Content-Type be for a 4xx error without a body?](https://stackoverflow.com/questions/15032932/what-should-the-content-type-be-for-a-4xx-error-without-a-body)

This differs from real blob storage and legacy emulator
### Steps to reproduce the issue?
HEAD http://127.0.0.1:10000/devstoreaccount1/mycontainer/boo.bak?st=<...>
### Have you found a mitigation/solution?
Not really. We could change the product code to ignore such case, but we'd rather have same code path for emulator and real storage.

**Comment by blueww** [comment]
@ayeltsov 
Thanks for reporting this issue!
Will look into it, and priority it with other Azurite feature request/bugs.

**Comment by ayeltsov** [comment]
More info: the root cause is that when we [create errors, we set content type and body](https://github.com/Azure/Azurite/blob/6d444ef1ab6f2933d63ca5fffa07de3abffa8f12/src/blob/errors/StorageError.ts#L52). However, we do that irrespective of the HTTP method (GET vs HEAD). However, per standard - [RFC 7231](https://tools.ietf.org/html/rfc7231#section-4.3.2), body should not be present for HEAD requests, so node simply drops it on the floor (silently, unless NODE_DEBUG is set).

I'm not sure what the right fix it - I simply hacked as proof of concept, but of course that can't be real fix since this is a generated file:

```
    var request = context.request;
    var method = (request == null || request == undefined) ? null : request.getMethod();

    if (method === 'HEAD') {
      console.log("It's a HEAD");
      logger.error(
        `ErrorMiddleware: Method is HEAD, skipping content type: ${err.contentType} and HTTP body: ${JSON.stringify(err.body)} `,
        context.contextId
      );
    }
    else {
      if (err.contentType) {
        logger.error(
          `ErrorMiddleware: Set content type: ${err.contentType}`,
          context.contextId
        );
        res.setContentType(err.contentType);
      }

      logger.error(
        `ErrorMiddleware: Set HTTP body: ${JSON.stringify(err.body)}`,
        context.contextId
      );
      if (err.body) {
        res.getBodyStream().write(err.body);
      }
    }
```

**Comment by blueww** [comment]
@ayeltsov 
Thanks for the detail information!
We will consider it when fixing this issue.

**Comment by XiaoningLiu** [comment]
Hi, thanks for your prototype! Fixed in v3.5.0. Will close this.
